Shared/Samples/Generate offline map with local basemap/GenerateOfflineMapWithLocalBasemapView.swift

```diff
@@ -0,0 +1,302 @@
+    
+    var body: some View {
+        GeometryReader { geometry in
+            MapViewReader { mapView in
+                MapView(map: model.offlineMap ?? model.onlineMap)
+                    .interactionModes(isGeneratingOfflineMap ? [] : [.pan, .zoom])
+                    .errorAlert(presentingError: $error)
+                    .task {
+                        do {
+                            try await model.initializeOfflineMapTask()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .onDisappear {
+                        Task { await model.cancelJob() }
+                    }
+                    .overlay {
+                        Rectangle()
+                            .stroke(.red, lineWidth: 2)
+                            .padding(EdgeInsets(top: 20, leading: 20, bottom: 44, trailing: 20))
+                            .opacity(model.offlineMap == nil ? 1 : 0)
+                    }
+                    .overlay {
+                        if isGeneratingOfflineMap,
+                           let progress = model.generateOfflineMapJob?.progress {
+                            VStack(spacing: 16) {
+                                ProgressView(progress)
+                                    .progressViewStyle(.linear)
+                                    .frame(maxWidth: 200)
+                                
+                                Button("Cancel") {
+                                    isCancellingJob = true
+                                }
+                                .disabled(isCancellingJob)
+                                .task(id: isCancellingJob) {
+                                    guard isCancellingJob else { return }
+                                    // Cancels the job.
+                                    await model.cancelJob()
+                                    // Sets cancelling the job and generating an
+                                    // offline map to false.
+                                    isCancellingJob = false
+                                    isGeneratingOfflineMap = false
+                                }
+                            }
+                            .padding()
+                            .background(.regularMaterial)
+                            .clipShape(RoundedRectangle(cornerRadius: 15))
+                            .shadow(radius: 3)
+                        }
+                    }
+                    .overlay(alignment: .top) {
+                        if model.offlineMap != nil {
+                            Text("Offline map generated.")
+                                .frame(maxWidth: .infinity, alignment: .center)
+                                .padding(8)
+                                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+                        }
+                    }
```

Comment:
```suggestion
                    .overlay(alignment: .top) {
                        Text("Offline map generated.")
                            .frame(maxWidth: .infinity, alignment: .center)
                            .padding(8)
                            .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
                            .opacity(model.offlineMap != nil ? 1 : 0)
                    }
```

Consider using `View.opacity(_:)` here instead to preserve the structural identity like you did in the red Rectangle overlay. Up to you.